x-pack/plugins/security_solution/public/detections/components/rules/data_view_selector/index.tsx

```diff
@@ -53,15 +54,26 @@ export const DataViewSelector = ({ kibanaDataViews, field }: DataViewSelectorPro
       : []
   );
 
+  const [showDataViewAlertsOnAlertsWarning, setShowDataViewAlertsOnAlertsWarning] = useState(false);
+
   useEffect(() => {
     if (!selectedDataViewNotFound && dataViewId) {
-      setSelectedOption([
-        { id: kibanaDataViews[dataViewId].id, label: kibanaDataViews[dataViewId].title },
-      ]);
+      const dataViewsTitle = kibanaDataViews[dataViewId].title;
+      const dataViewsId = kibanaDataViews[dataViewId].id;
+
+      setShowDataViewAlertsOnAlertsWarning(dataViewsId === 'security-solution-default');
```

Comment:
Nit: is there a subtle reason why this is implemented via `useEffect` when `showDataViewAlertsOnAlertsWarning` is a state derivative from `dataViewId`? Seems like could be implemented via `useMemo`.